File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/library/book/bookResponse.ts
import { AuthorResponse, Book, CoverType, GenreResponse, mapAuthorResponseToAuthor, mapGenreResponseToGenre, mapPublisherResponseToPublisher, PublisherResponse } from "../../../..";

export interface BookResponse {
    id: number;
    name: string;
    publicationDate: Date;
    price: number;
    coverType: CoverType;
    pageAmount: number;
    stockAmount: number;
    coverImgUrl: string,
    description: string | null,
    author: AuthorResponse;
    genre: GenreResponse;
    publisher: PublisherResponse;
}
export function mapBookResponseToBook(response: BookResponse): Book {
    return {
        id: response?.id,
        name: response?.name,
        publicationDate: new Date(response?.publicationDate),
        price: response?.price,
        coverType: response?.coverType,
        pageAmount: response?.pageAmount,
        stockAmount: response?.stockAmount,
        coverImgUrl: response?.coverImgUrl,
        description: response?.description,
        author: mapAuthorResponseToAuthor(response?.author),
        genre: mapGenreResponseToGenre(response?.genre),
        publisher: mapPublisherResponseToPublisher(response?.publisher),
    }
}

File: src/ELibrary.Backend/LibraryShopEntities/Domain/Dtos/Library/BookResponse.cs
using LibraryShopEntities.Domain.Entities.Library;

namespace LibraryShopEntities.Domain.Dtos.Library
{
    public class BookResponse
    {
        public int Id { get; set; }
        public string? Name { get; set; }
        public DateTime PublicationDate { get; set; }
        public decimal Price { get; set; }
        public CoverType CoverType { get; set; }
        public string? CoverImgUrl { get; set; }
        public int PageAmount { get; set; }
        public int StockAmount { get; set; }
        public string? Description { get; set; }
        public AuthorResponse? Author { get; set; }
        public GenreResponse? Genre { get; set; }
        public PublisherResponse? Publisher { get; set; }
    }
}